### PR TITLE
CP-15790, CP-17444

### PR DESCRIPTION
**Pull request by GaborApatiNagy**
Signed-off-by: Gabor Apati-Nagy <gabor.apati-nagy@citrix.com>

### CODE REVIEW

File: XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs [hunk 1]
```diff
@@ -83,10 +83,10 @@ public DownloadUpdatesXmlAction(bool checkForXenCenter, bool checkForServerVersi
             _checkForXenCenter = checkForXenCenter;
             _checkForServerVersion = checkForServerVersion;
             _checkForPatches = checkForPatches;
-            _checkForUpdatesUrl = string.IsNullOrEmpty(checkForUpdatesUrl) ? InvisibleMessages.XENSERVER_UPDATE_URL : checkForUpdatesUrl;
+            _checkForUpdatesUrl = checkForUpdatesUrl;
```
Comment (MihaelaStoica) [inline XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs]: The `IsNullOrEmpty `check is still needed for the case when this constructor is called with the last parameter null, like this usage:
`this(checkForXenCenter, checkForServerVersion, checkForPatches, null)`
on line 90 below.

Comment (GaborApatiNagy) [inline XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs]: Following a quick discussion on this, we agreed I will put an Assert in for null only.

Comment (GaborApatiNagy) [inline XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs]: Added, please see the most recent commit

**Comment by MihaelaStoica** [comment]
Looks gooooood now. Merging

**Comment by GaborApatiNagy** [comment]
Thank youuuuuuu

**Pull request merged**